configs/experiment/im2im/segmentation_plugin.yaml

```diff
@@ -41,6 +44,8 @@ trainer:
 
 data:
   path: MUST_OVERRIDE # string path to manifest
+  cache_dir: ${paths.output_dir}/cache # string path to cache_dir (this speeds up data loading)
+  num_workers: MUST_OVERRIDE # this should be set based on the number of available CPU cores
```

Comment:
If this is a must, and should always be based on cores, could cyto-dl query the system for # cores?

Reply:
I think it makes sense to allow an override
- For the plugin, preprocessing is minimal, datasets are relatively small, and I think our target audience doesn't have crazy cpu numbers. This all limits the memory overhead so I'm not worried about using all cpus for this use case
- For more complicated training on larger systems (e.g. A100s with 256 cpus), automatically setting 256 workers will use all available RAM/ be incredibly slow. Using > 16 workers seems to cause freezing instead of a further speedup.

Reply:
What do you think about setting a default `num_workers`? Maybe to 1. Would that be very slow?
@hughes036 since this is a required override we need to make the related change https://github.com/AllenCell/allencell-ml-segmenter/issues/290 on ml-segmenter

Reply:
It seems like one way to do this would be on segmenter's side, override `num_workers` to be something like `min(NUM_USER_CPUS, 16)`?